### ISSUE TITLE
bindgen can't generate thread local variables correctly

### ISSUE THREAD

**Issue by framlog** [rust-lang-nursery/rust-bindgen#1187]
I don't know whether it's really an issue since stable rust doesn't support thread_local yet.

### Input C/C++ Header

```C++
static __thread int variable_1 = 0;
thread_local int variable_2 = 0;
```

### Bindgen Invocation

```Rust
bindgen::Builder::default()
    .header("input.h")
    .rust_target(bindgen::RustTarget::Nightly)
    .generate()
    .unwrap()
```

or

```
$ bindgen input.h --whatever --flags
```

### Actual Results

```rust
pub static mut variable_1 ...
```

### Expected Results

```rust
#[thread_local]
pub static mut variable_1
```

**Comment by emilio** [comment]
There's another issue, I think, which is that thread-local vars don't have symbols, so I'm not sure it's possible to touch a C thread-local variable without going through FFI.

There's definitely a bug in there, if I'm right we shouldn't be generating TLS vars at all anyway.

**Comment by fintelia** [comment]
I'm also impacted by this issue. Is it possible to work around by blacklisting?

**Comment by emilio** [comment]
Yes, you should be able to just whitelist it, but I don't think you can provide the rust definition, given I don't think there's a standard way to create TLS vars across languages
